src/github/logic.py

```diff
@@ -277,10 +277,14 @@ def maybe_automerge_pull_request_and_rerun_stale_checks(
             and pull_request.is_mergeable()
             and pull_request.is_approved()
         )
-        did_rerun_stale_required_checks = (
-            is_pull_request_ready_for_automerge
-            and _maybe_rerun_stale_checks(pull_request)
-        )
+        # if PR has already rerun checks, we don't need to rerun them again
+        if did_rerun_checks:
+            did_rerun_stale_required_checks = did_rerun_checks
+        else:
+            did_rerun_stale_required_checks = (
+                is_pull_request_ready_for_automerge
+                and _maybe_rerun_stale_checks(pull_request)
+            )
```

Comment:
Here's a contentious suggestion:
```suggestion
        if not did_rerun_stale_required_checks:
            did_rerun_stale_required_checks = (
                is_pull_request_ready_for_automerge
                and _maybe_rerun_stale_checks(pull_request)
            )
```
aliasing parameters/variables is not a great pattern because it makes it harder to find these references in the future (for refactoring or just making changes to logic). 
However, overwriting parameters within the scope of the function are also not ideal because it's bad practice in every other language that does pass by reference.

I would prefer to trade the former over the latter but it's up to you